cylc/flow/data_store_mgr.py

```diff
@@ -1543,13 +1543,18 @@ def _apply_broadcasts_to_runtime(self, tokens, rtconfig):
             poverride(rtconfig, overrides, prepend=True)
         return rtconfig
 
-    def insert_job(self, name, cycle_point, status, job_conf):
+    def insert_job(
```

Comment:
This method is only used by `TaskEventsMgr._insert_task_job`. This function looks like it is designed to take a job status, but `_insert_task_job` has been feeding it the task status. 

This may be ok if the task status is also a job status, but the job will not be added to the data store if the status is not a job status, because of the check at line 1578.

In that case, subsequent calls to `delta_job_status` fail, because the data store doesn't have a record of the job.

In any case, `delta_job_status` is always subsequently called, so it doesn't matter if we don't set the job status here.